### ISSUE TITLE
Rich presence not working

### ISSUE THREAD

**Issue by knhs-2** [goaaats/FFXIVQuickLauncher#62]
Just saw that it was fixed in a recent update but it's still not working for me. I'm wondering if this is an issue on my end or if it's still a bit buggy

**Comment by goaaats** [comment]
Can you go to ``%AppData%\XIVLauncher\addons\RichPresence`` and upload the RichPresenceException.txt file that should be in there?

**Comment by knhs-2** [comment]
Well, I guess that would be the problem. No such folder exists. I'm not seeing the ability to add rich presence regardless

**Comment by goaaats** [comment]
Go to Settings -> Addons and tick the box next to Rich Presence

**Comment by knhs-2** [comment]
![da6686bd36b32142ef0f1e1f9befdd8f](https://user-images.githubusercontent.com/52429317/60549245-28ed7e00-9cd9-11e9-8127-5c11a8be4440.png)

**Comment by goaaats** [comment]
Can you post a screenshot of the about tab?

**Comment by knhs-2** [comment]
![5cd9f7cadf123528c4902b50083501d4](https://user-images.githubusercontent.com/52429317/60599224-f1c3af00-9d62-11e9-8340-6465210650f2.png)

**Comment by Snekks** [comment]
It's currently not working for me, either. It was working in the previous version before this recent update. Also, as a side note plugins are also not being loaded.

**Comment by goaaats** [comment]
Can you check ``%AppData%\XIVLauncher`` and upload the output.log

**Comment by knhs-2** [comment]
[output.log](https://github.com/goaaats/FFXIVQuickLauncher/files/3357080/output.log)

**Comment by goaaats** [comment]
Alright thanks, I'll check it out

**Comment by Snekks** [comment]
My logs are identical. Plugins are also not outputting their init messages to the in game chat as well if that helps.

**Comment by ChakraaThePanda** [comment]
Personally, it works.
The only thing missing in the RichPresence that I could see so far is Lakeland.
When you go in Lakeland, we lose the image and our job.

**Comment by DAOWAce** [comment]
Not working here either.
/xlhelp ingame doesn't do anything either.

Seems it's just not hooking whatsoever?

log:
 ```
2019-07-06 11:38:20.094 -04:00 [INF] XIVLauncher started with version 2.6.0.0, commit bdd1e1b
2019-07-06 11:38:20.672 -04:00 [INF] Loaded UI theme resource.
2019-07-06 11:38:20.672 -04:00 [INF] Loading MainWindow for account ''
2019-07-06 11:38:21.909 -04:00 [INF] Starting update check.
2019-07-06 11:38:22.284 -04:00 [INF] AutoUpdaterOnCheckForUpdateEvent called.
2019-07-06 11:38:22.284 -04:00 [INF] No update: 2.6.0.0
2019-07-06 11:38:24.146 -04:00 [INF] MainWindow initialized.
2019-07-06 11:38:51.340 -04:00 [INF] StartGame() called
2019-07-06 11:38:51.534 -04:00 [INF] Cache is invalid or disabled, logging in normally.
2019-07-06 11:38:53.877 -04:00 [INF] Starting game process: V:\Games\Steam\steamapps\common\FINAL FANTASY XIV - A Realm Reborn/game/ffxiv_dx11.exe
```

Starting richpresence exe manually: 

```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at FFXIVRichPresenceRunner.Program.Main(String[] args)
```

**Comment by DAOWAce** [comment]
Update: Upon deleting the XIVLauncher folders in %appdata%, the program doesn't restore anything in there, it only writes to the output.log.

The first version I ever used was 2.5 and it downloaded the addons/plugins folder, but I never actually checked if rich presence was working.  But now with 2.6, it seems it's just not downloading anything.

Seeing as the program force updates to the latest version, we can't rollback to an earlier install, and I didn't backup the folder prior to deleting it so I can't test anything further.

**Comment by DJScias** [comment]
Hello there,

I've been using the launcher for a few days but every time I run it and try the Rich Presence it doesn't really well.. Do much. I have noticed it does generate a `RichPresenceException.txt` file (regardless if I run the launcher as admin or not) that has the following in it:

```
System.ComponentModel.Win32Exception (0x80004005): Only part of a ReadProcessMemory or WriteProcessMemory request was completed
   at System.Diagnostics.NtProcessManager.GetModuleInfos(Int32 processId, Boolean firstModuleOnly)
   at System.Diagnostics.NtProcessManager.GetFirstModuleInfo(Int32 processId)
   at System.Diagnostics.Process.get_MainModule()
   at Nhaama.FFXIV.Actor.ActorTableCollection.Update()
   at Nhaama.FFXIV.Game.Update()
   at FFXIVRichPresenceRunner.Program.<Run>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

**Comment by DAOWAce** [comment]
The new version seems to have fixed Rich Presence for me.

Haven't checked anything else yet, but the advanced status in Discord is displaying now, so there's a good chance it's fixed. Hopefully others can chime in.

**Comment by DAOWAce** [comment]
Or it can completely break again..

```
System.Exception: Could not get definitions for version: https://raw.githubusercontent.com/goaaats/Nhaama/master/definitions/FFXIV/dx11/2019.07.10.0001.0000.json ---> System.Net.WebException: The remote server returned an error: (404) Not Found.
   at System.Net.WebClient.DownloadDataInternal(Uri address, WebRequest& request)
   at System.Net.WebClient.DownloadString(Uri address)
   at Nhaama.FFXIV.Definitions.Get(NhaamaProcess p, String version, GameType gameType)
   --- End of inner exception stack trace ---
   at Nhaama.FFXIV.Definitions.Get(NhaamaProcess p, String version, GameType gameType)
   at Nhaama.FFXIV.Game..ctor(Process process, Boolean loadRemote)
   at FFXIVRichPresenceRunner.Program.<Run>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

**Comment by DJScias** [comment]
@DAOWAce

I'm going to guess that might be because of the new patch (5.01) hitting and there not being definitions for it yet.

**Comment by goaaats** [comment]
I'm not at home for a few days so I probably won't be able to fix it for the newest patch, sorry

**Comment by JamieAzure** [comment]
Any news on this being resolved?

**Comment by NordicSnow** [comment]
it worked before, but i guess the new patch broke it again

**Comment by Zyian** [comment]
Is this still an issue? This was an issue with 2.6, now it's up to 2.9

**Comment by goaaats** [comment]
I'm rewriting the add-on system for 3.0, might fix that for you - shouldn't take too long anymore

**Comment by hlouzan** [comment]
I managed to make it work executing the launcher as admin, otherwise it doesn't work.

**Comment by DAOWAce** [comment]
Yeah, while it worked on and off previously, the feature after the re-write seems to be 100% issue free.

If the launcher now needs admin to work properly, then, well, that's an issue for non-power user systems, but otherwise everything looks good and this seems to be fully resolved.

**Comment by goaaats** [comment]
It should work fine without admin, at least on my system - what's your setup?

**Comment by hlouzan** [comment]
Tbh its kinda strange, the user account has Admin privileges it shouldn't need to be specified to be run as Admin, but Discord has the same problem when full-screen applications so =/ Its a Windows 10 Pro.

**Comment by JamieAzure** [comment]
I finally worked out what it was... I looked in the event log and saw it
was having a dx9 conflict.  Started shutting down apps on my PC until it
worked.  And now as I'm typing this I forgot which one it was. (pretty sure
it was either Riva Server or MSI Afterburner.  I'll test more later.

-----------------------------------------------------------------------------------------
*Jamie **Azure** Martin*
*Lead Application Developer*
*ColdFusion, PHP, Perl, jQuery, Javascript, CSS2/3 etc....*

*Enlightenment is free.  Accepting it will cost you.*
-----------------------------------------------------------------------------------------

**Comment by Algraud** [comment]
I am also having a RichPresence error, though its a different one:
`2019-12-01 14:32:03.127 +02:00 [ERR] Critical error in RichPresence.
Newtonsoft.Json.JsonSerializationException: Error converting value 29329202 to type 'Nhaama.Memory.Pointer'. Path 'TerritoryType', line 26, position 27. ---> System.ArgumentException: Could not cast or convert from System.Int64 to Nhaama.Memory.Pointer.
   at Newtonsoft.Json.Utilities.ConvertUtils.EnsureTypeAssignable(Object value, Type initialType, Type targetType)
   at Newtonsoft.Json.Utilities.ConvertUtils.ConvertOrCast(Object initialValue, CultureInfo culture, Type targetType)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)
   --- End of inner exception stack trace ---
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at Nhaama.Memory.Serialization.Converters.PointerPropertyConverter.ReadJson(JsonReader reader, Type objectType, Object existingValue, JsonSerializer serializer)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.DeserializeConvertable(JsonConverter converter, JsonReader reader, Type objectType, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonConverter[] converters)
   at Nhaama.FFXIV.Definitions.Get(NhaamaProcess p, String version, GameType gameType)
   at Nhaama.FFXIV.Game..ctor(Process process, Boolean loadRemote)
   at XIVLauncher.Addon.RichPresenceAddon.<DoWork>d__5.MoveNext()`

**Comment by Zyian** [comment]
> 
> 
> I am also having a RichPresence error, though its a different one:
> 
> ```
> 2019-12-01 14:32:03.127 +02:00 [ERR] Critical error in RichPresence.
> Newtonsoft.Json.JsonSerializationException: Error converting value 29329202 to type 'Nhaama.Memory.Pointer'. Path 'TerritoryType', line 26, position 27. ---> System.ArgumentException: Could not cast or convert from System.Int64 to Nhaama.Memory.Pointer.
>    at Newtonsoft.Json.Utilities.ConvertUtils.EnsureTypeAssignable(Object value, Type initialType, Type targetType)
>    at Newtonsoft.Json.Utilities.ConvertUtils.ConvertOrCast(Object initialValue, CultureInfo culture, Type targetType)
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)
>    --- End of inner exception stack trace ---
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
>    at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
>    at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
>    at Nhaama.Memory.Serialization.Converters.PointerPropertyConverter.ReadJson(JsonReader reader, Type objectType, Object existingValue, JsonSerializer serializer)
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.DeserializeConvertable(JsonConverter converter, JsonReader reader, Type objectType, Object existingValue)
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
>    at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
>    at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
>    at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonConverter[] converters)
>    at Nhaama.FFXIV.Definitions.Get(NhaamaProcess p, String version, GameType gameType)
>    at Nhaama.FFXIV.Game..ctor(Process process, Boolean loadRemote)
>    at XIVLauncher.Addon.RichPresenceAddon.<DoWork>d__5.MoveNext()
> ```

Echoing this error, currently on `xivlauncher-4.1.0.0-89ca295-dirty` of the launcher

**Comment by goaaats** [comment]
You may need to pull lib/Nhaama master to get it to work, we merged a PR and I didn't get around to patch the launcher yet - will revert soon till I can work on it

**Comment by Aida-Enna** [comment]
Fixed with #120, closing. Thank you!
